sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -205,27 +209,32 @@ case class ConcatWs(children: Seq[Expression])
       }.unzip
 
       val codes = ctx.splitExpressions(ctx.INPUT_ROW, evals.map(_.code))
-      val varargCounts = ctx.splitExpressions(varargCount, "varargCountsConcatWs",
-        ("InternalRow", ctx.INPUT_ROW) :: Nil,
-        "int",
-        { body =>
+      val varargCounts = ctx.splitExpressions(
+        expressions = varargCount,
+        funcName = "varargCountsConcatWs",
+        arguments = ("InternalRow", ctx.INPUT_ROW) :: Nil,
+        returnType = "int",
+        makeSplitFunction = { body =>
           s"""
            int $varargNum = 0;
            $body
            return $varargNum;
          """
         },
-        _.mkString(s"$varargNum += ", s";\n$varargNum += ", ";"))
-      val varargBuilds = ctx.splitExpressions(varargBuild, "varargBuildsConcatWs",
-        ("InternalRow", ctx.INPUT_ROW) :: ("UTF8String []", array) :: ("int", idxInVararg) :: Nil,
-        "int",
-        { body =>
+        foldFunctions = _.mkString(s"$varargNum += ", s";\n$varargNum += ", ";"))
+      val varargBuilds = ctx.splitExpressions(
+        expressions = varargBuild,
+        funcName = "varargBuildsConcatWs",
+        arguments =
+          ("InternalRow", ctx.INPUT_ROW) :: ("UTF8String []", array) :: ("int", idxInVararg) :: Nil,
+        returnType = "int",
+        makeSplitFunction = { body =>
```

Comment:
nit: is it possible to eliminate the `{  }`?

Reply:
done.